bin/generate_builtins.jl

```diff
@@ -137,47 +138,79 @@ function maybe_evaluate_builtin(frame, call_expr, expand::Bool)
             print(io,
 """
     $head f === tuple
-        return Some{Any}(ntuple(i->@lookup(frame, args[i+1]), length(args)-1))
+        return Some{Any}(ntupleany(i->@lookup(frame, args[i+1]), length(args)-1))
 """)
             continue
-        elseif f === Core._apply || f === Core._apply_latest
-            # Resolve varargs calls and invokelatest
-            fstr = scopedname(f)
+        elseif f === Core._apply_iterate
+            # Resolve varargs calls
             print(io,
 """
-    $head f === $fstr
-        args = getargs(args, frame)
+    $head f === Core._apply_iterate
+        argswrapped = getargs(args, frame)
         if !expand
-            return Some{Any}($fstr(args...))
+            return Some{Any}(Core._apply_iterate(argswrapped...))
         end
-        new_expr = Expr(:call, args[1])
-        popfirst!(args)
-        for x in args
-            push!(new_expr.args, (isa(x, Symbol) || isa(x, Expr) || isa(x, QuoteNode)) ? QuoteNode(x) : x)
+        aw1 = argswrapped[1]::Function
+        @assert aw1 === Core.iterate || aw1 === Core.Compiler.iterate || aw1 === Base.iterate "cannot handle `_apply_iterate` with non iterate as first argument, got \$(aw1), \$(typeof(aw1))"
+        new_expr = Expr(:call, argswrapped[2])
+        popfirst!(argswrapped) # pop the iterate
+        popfirst!(argswrapped) # pop the function
+        argsflat = append_any(argswrapped...)
+        for x in argsflat
+            push!(new_expr.args, QuoteNode(x))
         end
-        return new_expr
+        return maybe_recurse_expanded_builtin(frame, new_expr)
 """)
             continue
         elseif f === Core.invoke
+            fstr = scopedname(f)
             print(io,
 """
-    $head f === invoke
+    $head f === $fstr
             argswrapped = getargs(args, frame)
             if !expand
-                return Some{Any}(invoke(argswrapped...))
+                return Some{Any}($fstr(argswrapped...))
             end
-            return Expr(:call, invoke, argswrapped...)
+            return Expr(:call, $fstr, argswrapped...)
 """)
             continue
+        elseif f === Core._call_latest
+            print(io,
+"""
+    elseif f === Core._call_latest
+        args = getargs(args, frame)
+        if !expand
+            return Some{Any}(Core._call_latest(args...))
+        end
+        new_expr = Expr(:call, args[1])
+        popfirst!(args)
+        for x in args
+            push!(new_expr.args, QuoteNode(x))
+        end
+        return maybe_recurse_expanded_builtin(frame, new_expr)
+""")
         end
 
         id = findfirst(isequal(f), Core.Compiler.T_FFUNC_KEY)
         fcall = generate_fcall(f, Core.Compiler.T_FFUNC_VAL, id)
-        print(io,
+        if f in Core.Builtin[
+            Core._call_in_world_total, Core.donotdelete,
```

Comment:
This requires Julia 1.9 to run the file now. Maybe that is ok.

Reply:
Yeah, I don't think there really is a way around that if we want the list to be up-to-date. I can maybe add a comment to that regard though